services/diagnostic/handlers.go

```diff
@@ -0,0 +1,1222 @@
+	return &KapacitorHandler{
+		l: h.l.With(zap.String("node", node)),
+	}
+}
+
+func (h *KapacitorHandler) WithEdgeContext(task, parent, child string) kapacitor.EdgeDiagnostic {
+	return &KapacitorHandler{
+		l: h.l.With(zap.String("task", task), zap.String("parent", parent), zap.String("child", child)),
+	}
+}
+
+func (h *KapacitorHandler) TaskMasterOpened() {
+	h.l.Info("opened task master")
+}
+
+func (h *KapacitorHandler) TaskMasterClosed() {
+	h.l.Info("closed task master")
+}
+
+func (h *KapacitorHandler) StartingTask(task string) {
+	h.l.Debug("starting task", zap.String("task", task))
+}
+
+func (h *KapacitorHandler) StartedTask(task string) {
+	h.l.Info("started task", zap.String("task", task))
+}
+
+func (h *KapacitorHandler) StoppedTask(task string) {
+	h.l.Info("stopped task", zap.String("task", task))
+}
+
+func (h *KapacitorHandler) StoppedTaskWithError(task string, err error) {
+	h.l.Error("failed to stop task with out error", zap.String("task", task), zap.Error(err))
+}
+
+func (h *KapacitorHandler) TaskMasterDot(d string) {
+	h.l.Debug("listing dot", zap.String("dot", d))
+}
+
+func (h *KapacitorHandler) ClosingEdge(collected int64, emitted int64) {
+	h.l.Debug("closing edge", zap.Int64("collected", collected), zap.Int64("emitted", emitted))
+}
+
+func (h *KapacitorHandler) Error(msg string, err error, ctx ...keyvalue.T) {
+	Error(h.l, msg, err, ctx)
+}
+
+func (h *KapacitorHandler) AlertTriggered(level alert.Level, id string, message string, rows *models.Row) {
+	h.l.Debug("alert triggered",
+		zap.Stringer("level", level),
+		zap.String("id", id),
+		zap.String("event_message", message),
+		zap.String("data", fmt.Sprintf("%v", rows)),
+	)
+}
+
+func (h *KapacitorHandler) SettingReplicas(new int, old int, id string) {
+	h.l.Debug("setting replicas",
+		zap.Int("new", new),
+		zap.Int("old", old),
+		zap.String("event_id", id),
+	)
+}
+
+func (h *KapacitorHandler) StartingBatchQuery(q string) {
+	h.l.Debug("starting next batch query", zap.String("query", q))
+}
+
+func TagPairs(tags models.Tags) []string {
+	ts := []string{}
+	for k, v := range tags {
+		ts = append(ts, k+"="+v)
+	}
+
+	return ts
+}
+
+func FieldPairs(tags models.Fields) []string {
+	ts := []string{}
+	for k, v := range tags {
+		ts = append(ts, fmt.Sprintf("%s=%v", k, v))
+	}
+
+	return ts
+}
+
+// TODO: deal with prefix and level
+func (h *KapacitorHandler) LogPointData(level, prefix string, point edge.PointMessage) {
+
+	h.l.Info(prefix,
+		zap.String("name", point.Name()),
+		zap.String("db", point.Database()),
+		zap.String("rp", point.RetentionPolicy()),
+		zap.String("group", string(point.GroupID())),
+		zap.Strings("dimensions", point.Dimensions().TagNames),
+		zap.Strings("tags", TagPairs(point.Tags())),
+		zap.Strings("fields", FieldPairs(point.Fields())),
+		zap.Time("time", point.Time()),
+	)
+}
+
+// TODO: deal with prefix and level
+func (h *KapacitorHandler) LogBatchData(level, prefix string, batch edge.BufferedBatchMessage) {
```

Comment:
I think prefix can just be a String field. The meaning prefix will be wrong at that point, but the intent is to make the logs unique which a field will do fine.

Reply:
@nathanielc any idea what the field should be named?